### PR TITLE
HAWKULAR-184 - Fixed accounts, so that it accepts basic auth.

### PR DESCRIPTION
**Pull request by jpkrohling**
@lucasponce , can you please review this fix? It should solve your basic-auth problem. If it looks OK, I'll merge it.

### CODE REVIEW

**Comment by lucasponce** [comment]
I would suggest to add a couple of curl examples into the sample/README.adoc to have a simple end-to-end example ready.
It can help a lot.

**Comment by pilhuhn** [comment]
On 11 May 2015, at 17:18, Lucas Ponce wrote:

> I would suggest to add a couple of curl examples into the 
> sample/README.adoc to have a simple end-to-end example ready.
> It can help a lot.

+1

**Comment by lucasponce** [comment]
For example, I'm testing this:
$ curl -u jdoe:password -H "Content-Type: application/json" -X POST -d '{"name":"test"}' http://localhost:8080/hawkular-accounts-sample/samples
Could not find MessageBodyWriter for response object of type: org.hawkular.accounts.sample.entity.Sample of media type: application/octet-stream

And I'm not sure if my curl is bad or if there is something missing in the sample app.

**Comment by jpkrohling** [comment]
@lucasponce , Can you try adding the "-H 'Accept: application/json' " as well ?

**Comment by jpkrohling** [comment]
And agree with the curl examples :)

**Comment by lucasponce** [comment]
Apart from the curl suggestions (to make more easy to test it), it seems the error is gone and a simple curl can work with the sample project.

**Comment by jpkrohling** [comment]
@lucasponce , I updated the README to include your example, plus the Accept header.

File: sample/README.adoc [hunk 1]
```diff
@@ -24,6 +24,17 @@ $ mvn clean install -Pdistribution,sample
...
 
+Note that for your own component, you'll have to adapt the standalone.xsl to include your deployment under the
+"secure-deployments" section.
+
+To test the sample, you can perform the following curl call:
+
+[source,bash]
+---
+$ curl -u jdoe:password -H "Accept: application/json" -H "Content-Type: application/json" -X POST -d '{"name":"test"}'
+http://localhost:8080/hawkular-accounts-sample/samples
```
Comment (lucasponce) [inline sample/README.adoc:35]: PR looks good to me.
For the future, as curl examples are "cheap" we can add more use cases GET/POST/PUT/DELETE to show a global round.
Whatever is just copy/paste from an user perspective helps to use/jump into the code more easily.
But that is not a stopper from my side, please go ahead with the merge.
Thanks,

**Comment by jpkrohling** [comment]
Merged via CLI.

**Pull request closed without merging**